### PR TITLE
add quicklab setup script

### PR DESCRIPTION
**Pull request by scoheb**

### CODE REVIEW

File: hack/quicklab/setup-nfs-quicklab.sh [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  OCP=$(oc whoami --show-server=true)
+  if [ "${OCP}" != "${QOCP}" ];
+  then
+    echo "It seems you are logged in to the wrong OCP cluster (${OCP}). Please login to ${QOCP} and retry the script";
+    exit 1;
+  fi;
+fi
+
+# default to 1 PV
+[[ -z $2 ]] && PVN="1" || PVN=$3
```
Comment (lcarva) [inline hack/quicklab/setup-nfs-quicklab.sh:38]: Should the end be `PVN=$2` ?

File: hack/quicklab/setup-nfs-quicklab.sh [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+set +e
+oc create namespace openshift-nfs-storage
+oc label namespace openshift-nfs-storage "openshift.io/cluster-monitoring=true" --overwrite=true
+oc project openshift-nfs-storage
+NAMESPACE=`oc project -q`
+oc create -f templates/rbac.yaml
+oc adm policy add-scc-to-user hostmount-anyuid system:serviceaccount:$NAMESPACE:nfs-client-provisioner
+echo -e "$(eval "echo -e \"$(<templates/deployment.yaml)\"")"|oc create -f -
+sleep 10
```
Comment (lcarva) [inline hack/quicklab/setup-nfs-quicklab.sh:81]: Waiting for the operator to come up? Maybe try this: `oc -n $NAMESPACE wait --for=condition=ready pod --all`

File: hack/quicklab/setup-nfs-quicklab.sh [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+#START AND ENABLE RPCBIND AND NFS SERVICES
+$SSH -C "sudo systemctl restart rpcbind && sudo systemctl restart nfs"
+
+set +e
+oc create namespace openshift-nfs-storage
+oc label namespace openshift-nfs-storage "openshift.io/cluster-monitoring=true" --overwrite=true
+oc project openshift-nfs-storage
+NAMESPACE=`oc project -q`
+oc create -f templates/rbac.yaml
```
Comment (lcarva) [inline hack/quicklab/setup-nfs-quicklab.sh:78]: `oc apply` brings the script closer to being idempotent. Same when creating the storage class.

File: hack/quicklab/setup-nfs-quicklab.sh [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+oc adm policy add-scc-to-user hostmount-anyuid system:serviceaccount:$NAMESPACE:nfs-client-provisioner
+echo -e "$(eval "echo -e \"$(<templates/deployment.yaml)\"")"|oc create -f -
+sleep 10
+oc create -f templates/storageClass.yaml
+oc patch storageclass managed-nfs-storage -p '{"metadata": {"annotations": {"storageclass.kubernetes.io/is-default-class": "true"}}}'
```
Comment (lcarva) [inline hack/quicklab/setup-nfs-quicklab.sh:83]: Could we just add this to `templates/storageClass.yaml` ?

File: hack/quicklab/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
+# Configuring NFS storage provisioner on Quicklab clusters
```
Comment (lcarva) [inline hack/quicklab/README.md:1]: It would be good to mention here that this install an operator, and link to the upstream project.

**Review by lcarva — COMMENTED** [review]
This is great! Thanks for making this PR.

File: hack/quicklab/README.md (hunk 5 above)
Comment (scoheb) [reply hack/quicklab/README.md:1]: added references and an explanation.

File: hack/quicklab/setup-nfs-quicklab.sh (hunk 4 above)
Comment (scoheb) [reply hack/quicklab/setup-nfs-quicklab.sh]: updated yaml file

File: hack/quicklab/setup-nfs-quicklab.sh (hunk 3 above)
Comment (scoheb) [reply hack/quicklab/setup-nfs-quicklab.sh]: switched to `apply`

File: hack/quicklab/setup-nfs-quicklab.sh (hunk 1 above)
Comment (scoheb) [reply hack/quicklab/setup-nfs-quicklab.sh]: fixed

File: hack/quicklab/setup-nfs-quicklab.sh (hunk 2 above)
Comment (scoheb) [reply hack/quicklab/setup-nfs-quicklab.sh]: much better! thanks

**Review by lcarva — APPROVED** [review]

**Comment by Michkov** [comment]
Could be issue in future that upi host has only one disk with 20 GB, 18 GB available.

**Comment by lcarva** [comment]
> Could be issue in future that upi host has only one disk with 20 GB, 18 GB available.

Very much so, but the benefits outweigh this issue IMO.

File: hack/quicklab/setup-nfs-quicklab.sh [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+echo 'cp /etc/exports ./exports; for i in '${DIRS}'; do  echo "$i *(no_root_squash,rw,sync)" >> ./exports; done; sudo cp ./exports /etc/exports; rm ./exports'|$SSH
+
+#START AND ENABLE RPCBIND AND NFS SERVICES
+$SSH -C "sudo systemctl restart rpcbind && sudo systemctl restart nfs"
+
+set +e
+oc create namespace openshift-nfs-storage
+oc label namespace openshift-nfs-storage "openshift.io/cluster-monitoring=true" --overwrite=true
+oc project openshift-nfs-storage
+NAMESPACE=`oc project -q`
```
[the remark this answers is not in the document]
Comment (scoheb) [reply hack/quicklab/setup-nfs-quicklab.sh]: fixed

File: hack/quicklab/setup-nfs-quicklab.sh [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+if [ -z "$1" ]; then
+    echo "usage: setup-nfs-quicklab.sh <upi host name>"
+    exit 1
+fi
+
+REMOTE=$1
+QOCP="https://$(echo ${REMOTE} |sed s/upi-0/api/):6443"
+
+scp -o StrictHostKeyChecking=no -i $QUICKLABKEY quicklab@$REMOTE:/home/quicklab/oc4/auth/kubeconfig /tmp/kubeconfig
```
[the remark this answers is not in the document]
Comment (scoheb) [reply hack/quicklab/setup-nfs-quicklab.sh:23]: moved download of key to earlier in script

File: hack/quicklab/setup-nfs-quicklab.sh [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+wget -q https://gitlab.cee.redhat.com/cee_ops/quicklab/raw/master/docs/quicklab.key -O $QUICKLABKEY
+chmod 600 $QUICKLABKEY
+
+SSH="ssh -o StrictHostKeyChecking=no -i $QUICKLABKEY quicklab@$REMOTE"
+
+echo -e Adding $PVN nfs folders on $REMOTE using $QUICKLABKEY
+
+#ENABLE NFS PORTS ON DEFAULT FIREWALL
+export DZONE=$($SSH -C 'sudo firewall-cmd --get-default-zone')
+export ip=$($SSH -C "sudo ifconfig"|grep inet|head -1|xargs|cut -d ' ' -f2)
```
[the remark this answers is not in the document]
Comment (scoheb) [reply hack/quicklab/setup-nfs-quicklab.sh]: fixed

File: hack/quicklab/setup-nfs-quicklab.sh [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+oc apply -f $ROOT/hack/quicklab/templates/storageClass.yaml
+set -e
+
+unset NAMESPACE
+unset QUICKLABKEY
+unset REMOTE
+unset DZONE
+unset ip
+unset SSH
+unset PORT
```
Comment (psturc) [inline hack/quicklab/setup-nfs-quicklab.sh:92]: A minor thing. I'm getting
```bash
"unset PORT" command failed with exit code 0
```
`PORT` var is not exported - should it be? Also I see it has a value assigned twice in the script.

**Review by psturc — COMMENTED** [review]
I verified this against my cluster. It works great!

Just 2 minor comments

File: hack/quicklab/README.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
```
[the remark this answers is not in the document]
Comment (scoheb) [reply hack/quicklab/README.md:1]: will do!

**Comment by scoheb** [comment]
@psturc @Michkov updates made. tested locally. ready to merge.

**Pull request merged**